doc/options.md

```diff
@@ -0,0 +1,2865 @@
+
+Using the `repoRoot` argument is optional, but it has the advantage of not having to thread the standard arguments (especially `pkgs` and `inputs`) all over the place.
+
+### `inputs`
+
+Your flake inputs as defined in [`mkFlake.<in>.inputs`](#mkflakeininputs).
+
+Note that these `inputs` have been de-systemized against the current system.
+
+This means that you can use the following syntax:
+```nix
+inputs.n2c.packages.nix2container
+inputs.self.packages.foo
+```
+
+In addition to the usual syntax which mentions `system` explicitely.
+```nix 
+inputs.n2c.packages.x86_64-linux.nix2container
+inputs.self.packages.x86_64-darwin.foo
+```
+
+#### `pkgs`
+
+A `nixpkgs` instantiated against the current system (as found in `pkgs.stdenv.system`), for each of your [`mkFlake.<in>.systems`](#mkflakeinsystems), and overlaid with goodies from `haskell.nix` and `iohk-nix`. 
+
+A `nixpkgs` is also available at `inputs.nixpkgs.legacyPackages` but that should *not* be used because it doesn't have the required overlays.
+
+You may reference `pkgs` freely to get to the legacy packages.
+
+#### `system`
+
+This is just `pkgs.stdenv.system`, which is likely to be used often.
+
+#### `lib`
+
+This is just `pkgs.lib` plus the `iogx` attrset, which contains library functions and utilities.
+
+In here you will find the following: 
+```nix 
+lib.iogx.mkProject {}
+lib.iogx.mkShell {}
+```
+
+
+---
+
+### `mkFlake.<in>.repoRoot`
+
+**Type**: path
+
+**Default**: `null`
+
+
+**Example**: `./.`
+
+
+The root of your repository.
+
+If not set, this will default to the folder containing the `flake.nix` file, using `inputs.self`.
+
+
+---
+
+### `mkFlake.<in>.systems`
+
+**Type**: list of (one of "x86_64-linux", "x86_64-darwin", "aarch64-darwin", "aarch64-linux")
+
+**Default**: `[ "x86_64-linux" "x86_64-darwin" ]`
+
+
+
+
+The systems you want to build for.
+
+The [`mkFlake.<in>.outputs`](#mkflakeinoutputs) function will be called once for each system.
+
+
+---
+
+### `mkProject`
+
+**Type**: core API function
+
+
+
+**Example**: 
+```nix
+# project.nix 
+{ repoRoot, inputs, pkgs, lib, system }:
+let 
+  cabalProject = lib.iogx.mkProject {
+    mkShell = repoRoot.nix.make-shell;
+
+    readTheDocs.siteFolder = "doc/read-the-docs-site";
+    
+    combinedHaddock.enable = true;
+    
+    cabalProjectArgs = {
+
+      compiler-nix-name = "ghc8107";
+
+      flake.variants.FOO = {
+        compiler-nix-name = "ghc927";
+      };
+    };
+  };
+in 
+[
+  {
+    inherit cabalProject;
+  }
+  {
+    hydraJobs.FOO = cabalProject.projectVariants.FOO.iogx.hydraJobs;
+  }
+]
+
+```
+
+
+The `lib.iogx.mkProject` function takes an attrset of options and returns a `cabalProject` with the `iogx` overlay.
+
+In this document:
+  - Options for the input attrset are prefixed by `mkProject.<in>`.
+  - The returned attrset contains the attributes prefixed by `mkProject.<out>.iogx`.
+
+
+---
+
+### `mkProject.<in>.cabalProjectArgs`
```

Comment:
I thought the goal was to be able to provide the *result* of `cabalProject'` if needed?